### DISCUSSION TITLE
Toggle the full history

### DISCUSSION THREAD

**Discussion by aloispichler** [i3/i3#4892] [Ideas]
From the docs:
`# switch between the current and the previously focused workspace:`
`bindsym $mod+b workspace back_and_forth`.

So cycle through the workspaces 1, 2, 3, 4, 5.
Then **mod+b** will bring the user to the workspace 4,
and **mod+b+b** will bring the user back to workspace 5, but nothing more.

Firefox, e.g., has a more natural and more useful workflow for exploring the _full_ history using the combination ctrl+tab:
**Ctrl+tab** will bring the user to tab 4,
**ctrl+tab+tab** back to 3,
and **ctrl+tab+tab+tab** to 2, and so on.
To achieve i3s behavior use **ctrl+tab** and **ctrl+tab**.

Is there a similar feature in i3?
Thank you. I really do enjoy i3!

**Comment by Airblader** [comment]
No, there is not. You can implement this using the IPC, however, and I believe there are already projects that have done so (I don't have them on hand, though).

**Comment by adamws** [comment]
This would be my starting point https://gist.github.com/adamws/6109404639b554a3e18fac33cd1ca68f
Server listens on i3 workspace focus events collecting history and also listens on `/tmp/i3history-socket` for workspace switch request.
Client is used for binding `bindsym Mod1+Tab exec ~/.local/bin/i3-tabs-client`

Opening workspace 1, 2, 3, 4 and then `Mod1+Tab` cycles through this order: 3 -> 2 -> 1 -> 2 -> 3 -> 4 -> 3 and so on...

**Comment by okraits** [comment]
@adamws I wrote a script for something like that: https://github.com/altdesktop/i3ipc-python/blob/master/examples/i3-cycle-focus.py

**Reply by medwatt** [reply]
@okraits, do you have an script that prevents a window thats not supposed to be tabbed from being tabbed when opened in a workspace that has a window that's tabbed?

**Reply by okraits** [reply]
So you basically want to exclude specific windows from the focus cycle?

**Reply by medwatt** [reply]
@okraits, I don't know if that's how it's described. Let me give an example. i3 allows you to set the layout of an application to tabbed. Let's say you're on a workspace and open several instances of an app with a tabbed layout. All these instances are tabbed. This is fine. Now, open and app that is not defined to be tabbed. We would expect this app to open on a separate column. However, it gets tabbed as well. This is not the behaviour I expect. The behaviour I expect is for the app to open on a separate column. Do you have a script that address this?

**Comment by RyanGibb** [comment]
> This would be my starting point https://gist.github.com/adamws/6109404639b554a3e18fac33cd1ca68f Server listens on i3 workspace focus events collecting history and also listens on `/tmp/i3history-socket` for workspace switch request. Client is used for binding `bindsym Mod1+Tab exec ~/.local/bin/i3-tabs-client`
> 
> Opening workspace 1, 2, 3, 4 and then `Mod1+Tab` cycles through this order: 3 -> 2 -> 1 -> 2 -> 3 -> 4 -> 3 and so on...

I've fixed this up into https://github.com/RyanGibb/i3-workspace-history/ which some people might find useful
